TLSharp.Core/TelegramClient.cs

```diff
@@ -56,22 +58,30 @@ public Session Session
             if (string.IsNullOrEmpty(apiHash))
                 throw new MissingApiConfigurationException("API_HASH");
 
-            if (store == null)
-                store = new FileSessionStore();
+            this.store = store ?? new FileSessionStore();
+            this.sessionUserId = sessionUserId;
 
             this.apiHash = apiHash;
             this.apiId = apiId;
             this.handler = handler;
             this.dcIpVersion = dcIpVersion;
 
-            session = Session.TryLoadOrCreateNew(store, sessionUserId);
+            session = Session.TryLoadOrCreateNew(this.store, sessionUserId);
```

Comment:
can't we just remove this line above? session will be initialized again anyway in ConnectAsync

Reply:
unfortunately not, because the following like needs it.

to save one initialization of session we must add extra code in the Connect method